src/Cmp/Storage/Adapter/S3AWSAdapter.php

```diff
@@ -192,175 +228,197 @@ public function rename($path, $newpath, $overwrite = false)
     }
 
     /**
-     * Delete a file or directory.
-     *
-     * @param string $path
+     * @param $newpath
+     * @param $overwrite
      *
-     * @return bool True on success, false on failure
+     * @throws FileExistsException
      */
-    public function delete($path)
+    private function ensureWeCanWriteDestFile($newpath, $overwrite)
```

Comment:
we have exactly the same check in src/Cmp/Storage/Adapter/FileSystemAdapter.php 
maybe in future common checks between those two classes we can create a common class for check it, and avoid duplicate it.

Reply:
I had some doubts about this because it's only one function, but yes, may be  is a good idea prepare a trait to afford this duplicate function and future duplicate checks if any.